Tests/BijectiveDictionaryTests/BijectiveDictionaryTests.swift

```diff
@@ -299,4 +299,92 @@ func conflict() {
     #expect(dict.conflict(with: ("A", 1)) == .pair)
     #expect(dict.conflict(with: ("A", 3)) == .both(otherLeft: "C", otherRight: 1))
 }
+
+@Test
+func uniquingWithNoConflicts() {
+    let pairs = [("A", 1), ("B", 2), ("C", 3)]
+    var conflicts = [BijectiveDictionary<String, Int>.Conflict]()
+    
+    let dict = BijectiveDictionary(pairs) { pair, conflict  in
+        conflicts.append(conflict)
+        return pair
+    }
+    #expect(dict == ["A": 1, "B": 2, "C": 3])
+    #expect(conflicts == [])
+}
+
+@Test
+func uniquingWithConflictingRightValues() {
+    let pairs = [("A", 1), ("B", 1), ("C", 3), ("D", 4), ("E", 3)]
+    var conflicts = [BijectiveDictionary<String, Int>.Conflict]()
+    
+    let dict = BijectiveDictionary(pairs) { pair, conflict in
+        conflicts.append(conflict)
+        return (pair.left, -conflicts.count)
+    }
+    
+    #expect(dict == ["A": 1, "B": -1, "C": 3, "D": 4, "E": -2])
+    #expect(conflicts == [.right, .right])
+}
+
+@Test
+func uniquingWithConflictingLeftValues() {
+    let pairs = [(1, "A"), (1, "B"), (3, "C"), (4, "D"), (3, "E")]
+    var conflicts = [BijectiveDictionary<Int, String>.Conflict]()
+    
+    let dict = BijectiveDictionary(pairs) { pair, conflict in
+        conflicts.append(conflict)
+        return (-conflicts.count, pair.right)
+    }
+    
+    #expect(dict == [1: "A", -1: "B", 3: "C", 4: "D", -2: "E"])
+    #expect(conflicts == [.left, .left])
+}
+
+@Test
+func uniquingWithConflictingPairs() {
+    let pairs = [(1, 2), (1, 2), (3, 4), (3, 4)]
+    var conflicts = [BijectiveDictionary<Int, Int>.Conflict]()
+    
+    let dict = BijectiveDictionary(pairs) { _, conflict in
+        conflicts.append(conflict)
+        return (-conflicts.count, -conflicts.count)
+    }
+    
+    #expect(dict == [1: 2, -1: -1, 3: 4, -2: -2])
+    #expect(conflicts == [.pair, .pair])
+}
+
+@Test
+func uniquingWithNoResolve() {
```

Comment:
If we shifted the responsibility of resolving the conflict to the user, then we would no longer need to test for this. Which would be nice.

Reply:
I see what you are saying. However, as long as the closure returns a left-right pair, there is the possibility that the closure produces a pair that still has a conflict. In `Dictionary`, the closure just returns what value to use when there are two conflicting keys; there are three possibilities:
1. Returns the value associated with the existing key $k$.
2. Returns the value associated with the new occurrence of key $k$.
3. Returns an arbitrary value to associate with key $k$.

Now I will consider an example with `BijectiveDictionary` where the user is able to get around resolving the conflict. For instance, say a `BijectiveDictionary` is being constructed using this initializer. It is mid way through processing the sequence of left-right pairs and already contains $(a, b)$. Next, it encounters $(a, c)$ (which conflicts because of a duplicate left value). In the closure, it is possible to return $(z, b)$; this fixes the conflicting left value but creates a new conflict with the right value. This condition is not possible with `Dictionary` since no matter what value is returned, the conflict will be resolved.

Please let me know if this makes sense.